File: TaskAngular/src/app/model/employee.ts
import {EmployeeType} from "./employee-type";

export class Employee {
  id : number;
  name : string;
  employeeType:EmployeeType;
  imageUrl : string;


  constructor(id: number, name: string, employeeType: EmployeeType, imageUrl: string) {
    this.id = id;
    this.name = name;
    this.employeeType = employeeType;
    this.imageUrl = imageUrl;
  }
}

File: Task/demo/src/main/java/com/example/demo/model/Employee.java
package com.example.demo.model;


import javax.persistence.*;

@Entity
public class Employee {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String name;

    @ManyToOne
    @JoinColumn(name = "typeId", updatable = false)
    private EmployeeType employeeType;

    private String imageUrl;

    public Employee() {
    }


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public EmployeeType getEmployeeType() {
        return employeeType;
    }

    public void setEmployeeType(EmployeeType employeeType) {
        this.employeeType = employeeType;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }
}
